ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -935,8 +942,11 @@ else if(colTypeLowerCase.equals(serdeConstants.SMALLINT_TYPE_NAME)){
         cs.setNumTrues(Math.max(1, numRows/2));
         cs.setNumFalses(Math.max(1, numRows/2));
         cs.setAvgColLen(JavaDataModel.get().primitive1());
-    } else if (colTypeLowerCase.equals(serdeConstants.TIMESTAMP_TYPE_NAME) ||
-        colTypeLowerCase.equals(serdeConstants.TIMESTAMPLOCALTZ_TYPE_NAME)) {
+    } else if (colTypeLowerCase.equals(serdeConstants.TIMESTAMP_TYPE_NAME)) {
+      cs.setAvgColLen(JavaDataModel.get().lengthOfTimestamp());
+      // epoch, seconds since epoch
+      cs.setRange(0, 2177452799L);
```

Comment:
Is this 2038-12-31 11:59:59 PM GMT? Why is it hard coded to that? If there is a reason for that, then I suggest to introduce a more meaningful constant declaration, like this:

// Explanation why the timestamp range upper limit should be this date
private static final long TIMESTAMP_RANGE_UPPER_LIMIT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss zzz").parse("2038-12-31 23:59:59 GMT").getTime() / 1000;

Reply:
I took the corresponding seconds value for the value that had been hardcoded for `date` type. I am not sure about the original reason to choose that value but I thought that consistency was good in this case. I will add a comment.

Reply:
I do not think this is critical, but I explored a bit and this seems to be a poor choice for a heuristic, since in most cases it will lead to underestimation of the data size (since most users will not have dates starting from 1970).
I will take as lower limit `01-01-2015` (epoch for ORC) and as upper limit `12-31-2025`. Let me know if you see any cons with this approach.